google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/impl/GoogleSession.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.storage.impl
+
+import akka.annotation.InternalApi
+import akka.stream.Materializer
+import akka.stream.alpakka.googlecloud.storage.impl.GoogleTokenApi.AccessTokenExpiry
+
+import scala.concurrent.Future
+
+@InternalApi
+private[impl] class GoogleSession(clientEmail: String, privateKey: String, tokenApi: GoogleTokenApi) {
+  protected var maybeAccessToken: Option[Future[AccessTokenExpiry]] = None
+
+  private def getNewToken()(implicit materializer: Materializer): Future[AccessTokenExpiry] = {
+    val accessToken = tokenApi.getAccessToken(clientEmail = clientEmail, privateKey = privateKey)
+    maybeAccessToken = Some(accessToken)
+    accessToken
+  }
+
+  private def expiresSoon(g: AccessTokenExpiry): Boolean =
+    g.expiresAt < (tokenApi.now + 60)
+
+  def getToken()(implicit materializer: Materializer): Future[String] = {
```

Comment:
I'm using this lib in production for few months and I noticed that this call can fail, it is transient issue. Does it make sense to retry it?

Reply:
You are right, we should be careful with back off. 
What do you suggest for defaults?